TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -642,7 +646,29 @@ def makeTreeFromMiniAOD(
                                          'JetsProperties:neutralHadronMultiplicity(Jets_neutralHadronMultiplicity)',
                                          'JetsProperties:photonMultiplicity(Jets_photonMultiplicity)',
                                          'JetsProperties:flavor(Jets_flavor)'])
-
+    if IncludeBEff:
+    	process.bTaggingEff        = cms.EDAnalyzer('BTaggingEffAnalyzer',
+        	JetsTag            = cms.InputTag('HTJets'),
+        	DiscriminatorTag   = cms.string('pfCombinedInclusiveSecondaryVertexV2BJetTags'),
+        	DiscriminatorValue = cms.double(0.890),
+        	PtNBins            = cms.int32(100),
+        	PtMin              = cms.double(0.),
+        	PtMax              = cms.double(1000.),
+        	EtaNBins           = cms.int32(60),
+        	EtaMin             = cms.double(-3.),
+        	EtaMax             = cms.double(3.)
+    	)
+   	process.Baseline += process.bTaggingEff
+    if AddBtagSF:
+	process.BTagScale=cms.EDProducer('BTagScale',
+		JetsTag            = cms.InputTag('HTJets'),
+		BTagEffInput   =cms.string('./data/test_run_RA2AnalysisTree.root'),
```

Comment:
I'm still a bit confused about why we need to track this file. Do I understand correctly that there will be one of these for every model point in the signal scan? As I mentioned in the comment on BTaggingEffAnalyzer, it seems like this process is more appropriately placed post-ntuple production...

These uncertainties will only be used in the datacards for the signal samples, right? I think we could just calculate them on the fly when we make datacards.